### PR TITLE
Regression tests

### PR DESCRIPTION
**Pull request by staabm**
closes https://github.com/phpstan/phpstan/issues/7519
closes https://github.com/phpstan/phpstan-src/pull/1485

took the changes from https://github.com/phpstan/phpstan-src/pull/1485, so kudos to @drupol

### CODE REVIEW

**Comment by drupol** [comment]
Thanks !

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]
This issue is about false positive when returning from a method. So this needs ReturnTypeRuleTest case: https://github.com/phpstan/phpstan/issues/7519#issuecomment-1187671983

File: tests/PHPStan/Rules/Functions/ReturnTypeRuleTest.php [hunk 1]
```diff
@@ -181,4 +181,11 @@ public function testListWithNullablesUnchecked(): void
 		$this->analyse([__DIR__ . '/data/return-list-nullables.php'], []);
 	}
 
+	public function testBug7519(): void
+	{
+		$this->checkExplicitMixed = true;
+		$this->checkNullables = true;
+		$this->analyse([__DIR__ . '/data/bug-7519.php'], []);
```
Comment (ondrejmirtes) [inline tests/PHPStan/Rules/Functions/ReturnTypeRuleTest.php]: functions vs. methods :P Please be less sloppy.

**Review by ondrejmirtes — CHANGES_REQUESTED** [review]

File: tests/PHPStan/Rules/Functions/ReturnTypeRuleTest.php (hunk 1 above)
Comment (staabm) [reply tests/PHPStan/Rules/Functions/ReturnTypeRuleTest.php]: sorry. will take a break now.

**Comment by ondrejmirtes** [comment]
Thank you :) I don't trust a test I haven't seen fail, so part of writing a regression test is trying it out with a reverted commit that fixed the problem.

**Comment by staabm** [comment]
> so part of writing a regression test is trying it out with a reverted commit that fixed the problem.

I run the testfiles usually using an older phpstan release to verify they fail - and I did so here.
but did not see the Function vs. Method difference

**Comment by ondrejmirtes** [comment]
It couldn't fail for you if you tested method return type in `ReturnTypeRuleTest` for functions.

**Pull request merged**